.github/actions/deploy-before-production/action.yml

```diff
@@ -0,0 +1,40 @@
+name: Deploy Before Production
```

Comment:
This would be better staying in the build-and-deploy workflow. There's not much benefit from making this an action.